packages/sdk/src/evm/contracts/smart-contract.ts

```diff
@@ -273,6 +283,12 @@ export class SmartContract<TContract extends BaseContract = BaseContract>
     return undefined;
   }
 
+  private detectAppURI() {
+    if (detectContractFeature<AppURI>(this.contractWrapper, "AppURI")) {
+      return new ContractAppURI(this.contractWrapper);
```

Comment:
what did you decide regarding having the appURI inside ContractMetadata?

Reply:
I want to have that as a fallback option but felt too complex for v1. Will add it in the next release but wanted to get this sdk out there